### PR TITLE
Stojanovic/ep 103 in createedit read uijson and render labels hints

### PR DESCRIPTION
**Pull request by Ducica**

### CODE REVIEW

File: oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/components/I18nTextInputField/I18nTextInputField.jsx [hunk 1]
```diff
@@ -1,53 +1,46 @@
...
-    <GroupField fieldPath={fieldPath} optimized>
+    <GroupField fieldPath={fieldPath} optimized={optimized}>
       <LanguageSelectField
-        fieldPath={`${fieldPath}.lang`}
-        placeholder=""
-        required
+        fieldPath={lngFieldPath}
         width={lngFieldWidth}
         usedLanguages={usedLanguages}
+        {...getFieldData({ fieldPath: lngFieldPath, icon: "globe" })
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/components/I18nTextInputField/I18nTextInputField.jsx:31]: Shouldn't icon be just regular prop outside hook function?

File: oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/components/I18nRichInputField/I18nRichInputField.jsx [hunk 2]
```diff
@@ -1,62 +1,53 @@
...
   return (
-    <GroupField fieldPath={fieldPath} optimized>
+    <GroupField fieldPath={fieldPath} optimized={optimized}>
       <LanguageSelectField
-        fieldPath={`${fieldPath}.lang`}
-        required
+        fieldPath={lngFieldPath}
         width={lngFieldWidth}
         usedLanguages={usedLanguages}
+        {...getFieldData({ fieldPath: lngFieldPath, icon: "globe" })
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/components/I18nRichInputField/I18nRichInputField.jsx:33]: Shouldn't icon be just regular prop outside hook function?

File: oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/util.js [hunk 3]
```diff
@@ -21,6 +24,7 @@ export function parseFormAppConfig(rootElementId = "form-app") {
   const recordPermissions = getInputFromDOM("record-permissions");
   const files = getInputFromDOM("files");
   const links = getInputFromDOM("links");
+  formConfig.getFieldData = getFieldData(formConfig.ui_model);
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/util.js:27]: Why doing it like this, if FieldDataProvider does basically the same thing? https://github.com/oarepo/oarepo-ui/pull/206/files?diff=unified&w=1#diff-2beecbc1422ee475e5299c154a15418f2e8f6bad914a1800cd9881096bebc64bR28

File: oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/util.js [hunk 4]
```diff
@@ -56,13 +60,19 @@ export function createFormAppInit({
             <Router>
               <OverridableContext.Provider value={overrideStore.getAll()}>
                 <FormConfigProvider value={config}>
-                  <Overridable
-                    id={buildUID(overridableIdPrefix, "FormApp.layout")}
+                  <FieldDataProvider
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/util.js:63]: FieldDataProvider expect only `export const FieldDataProvider = ({ children, fieldPathPrefix = "" })` as props

File: oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/util.js [hunk 5]
```diff
@@ -81,6 +91,92 @@ export function createFormAppInit({
...
+        label: label,
+        placeholder: hint,
+        labelIcon: icon,
+        required,
+      },
+    };
+  };
+};
+
+export function transformPath(path) {
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/util.js:159]: function name doesn't tell much about what it's good for (without the need to study its body), I suggest renaming to `toModelPath(path)`

File: oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/util.js [hunk 6]
```diff
@@ -81,6 +91,92 @@ export function createFormAppInit({
...
+    } = _get(uiMetadata, path);
+    const label = i18next.t(modelLabel);
+    const help =
+      i18next.t(modelHelp) === modelHelp ? null : i18next.t(modelHelp);
+    const hint =
+      i18next.t(modelHint) === modelHint ? null : i18next.t(modelHint);
+    // full representation meaning jsx or small space representation (label with helptext in popup)
+    // text only without react elements
+    return {
+      fullRepresentation: {
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/util.js:121]: I would just return data needed and leave it to components calling this function, to render label component they want to.

Comment (mirekys) [reply oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/util.js:121]: Currently this returns 3 different options, creating 3 different label components, where only one of them gets used every time

File: oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/util.js (hunk 3 above)
Comment (Ducica) [reply oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/util.js:27]: @mirekys This is artifact from initial implementation, you are right. My bad

File: oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/components/I18nTextInputField/I18nTextInputField.jsx (hunk 1 above)
Comment (Ducica) [reply oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/components/I18nTextInputField/I18nTextInputField.jsx:31]: The icon is used by FieldLabel cmp (invenio) and for uniformity sake also for CompactLabel cmp (ours), which is why it is needed to return proper jsx

**Review by mirekys — APPROVED** [review]

**Review by mirekys — APPROVED** [review]

**Pull request merged**